File: KartMayhem.Model/Rezervacije.cs
using KartMayhem.Model.UniqueGetRequests;

namespace KartMayhem.Model
{
    public class Rezervacije
    {
        public int Id { get; set; }

        public int CijenaRezervacije { get; set; }

        public int BrojOsoba { get; set; }

        public string ImeStaze { get; set; }

        public string DayOfReservation { get; set; }

        public string TimeSlot { get; set; }

        public Korisnici Korisnik { get; set; }

        public Staze Staza { get; set; }
    }
}

File: UI/kartmayhem_mobile/lib/Models/rezervacije.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:kartmayhem_mobile/Models/staze.dart';

part 'rezervacije.g.dart';

/// dart run build_runner build

@JsonSerializable()
class Rezervacije {
  int? id;
  int? cijenaRezervacije;
  int? brojOsoba;
  Staze? staza;
  String? imeStaze;

  Rezervacije(
      {this.id,
      this.cijenaRezervacije,
      this.brojOsoba,
      this.staza,
      this.imeStaze});

  factory Rezervacije.fromJson(Map<String, dynamic> json) =>
      _$RezervacijeFromJson(json);
  Map<String, dynamic> toJson() => _$RezervacijeToJson(this);
}
